File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/CodeLlama-34b-Instruct-hf/Avatar/Python/Java-sanitized/codeforces_203_A.java
import java.util.Scanner;
import java.util.Queue;
import java.util.concurrent.Thread;

public class codeforces_203_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] xxs = input.split(" ");
        int initialValue = Integer.parseInt(xxs[0]);
        int t = Integer.parseInt(xxs[1]);
        int a = Integer.parseInt(xxs[2]);
        int b = Integer.parseInt(xxs[3]);
        int da = Integer.parseInt(xxs[4]);
        int db = Integer.parseInt(xxs[5]);

        Queue<String> queue = new Queue<>();
        Thread thread = new Thread(new Runnable() {
            @Override
            public void run() {
                String result = checkCombination(initialValue, t, a, b, da, db);
                queue.add(result);
            }
        });
        thread.start();
        thread.join();
        String result = queue.poll();
        if (result.equals("Result Not Found")) {
            System.out.println("NO");
        } else {
            System.out.println("YES");
        }
    }

    private static String checkCombination(int initialValue, int t, int a, int b, int da, int db) {
        if (initialValue == 0) {
            return "Result Found";
        }
        int aTime = Math.min(a / da, t - 1);
        int bTime = Math.min(b / db, t - 1);
        for (int i = 0; i <= aTime; i++) {
            for (int j = 0; j <= bTime; j++) {
                if (a - da * i == initialValue || b - db * j == initialValue) {
                    return "Result Found";
                }
                if (a - da * i + (b - db * j) == initialValue) {
                    return "Result Found";
                }
            }
        }
        return "Result Not Found";
    }
}

File: dataset/Intermediate/Translation/Avatar/v1-Python-code/codeforces_203_A.py
import numpy as np
import threading
import queue
xxs = input().split(' ')
(initial_value, t, a, b, da, db) = (int(xxs[0]), int(xxs[1]), int(xxs[2]), int(xxs[3]), int(xxs[4]), int(xxs[5]))

def check_combination():
    if initial_value == 0:
        print('YES')
        return 'Result Found'
    a_time = np.min(np.array([[a // da, t - 1]]))
    b_time = np.min(np.array([[b // db, t - 1]]))
    for i in range(a_time + 1):
        for j in range(b_time + 1):
            if a - da * i == initial_value or b - db * j == initial_value:
                print('YES')
                return 'Result Found'
            if a - da * i + (b - db * j) == initial_value:
                print('YES')
                return 'Result Found'
    return 'Result Not Found'
queue_main0 = queue.Queue()

def main_thread(queue):
    result = check_combination()
    queue.put(result)
thread_main0 = threading.Thread(target=main_thread, args=(queue_main0,))
thread_main0.start()
thread_main0.join()
result_main0 = queue_main0.get()
ff = result_main0
if ff == 'Result Not Found':
    print('NO')